pandas/core/frame.py

```diff
@@ -5693,6 +5693,8 @@ def _homogenize(data, index, dtype=None):
                 else:
                     v = dict(v)
                 v = lib.fast_multiget(v, oindex.values, default=NA)
+            elif v is None:
+                v = np.nan
```

Comment:
you can change the check where this breaks now (``if subarr.ndim = 0``) to
``if getattr(subarr, 'ndim', 0) == 0`` then it will fall into that expression where the ``None`` will be correctly inferred